nextjs-app/username-and-password/vite.config.mts

```diff
@@ -0,0 +1,29 @@
+import next from "next/vite";
+import { defineConfig } from "vite";
+
+export default defineConfig({
+	plugins: [
+		next({
+			plugins: [
+				{
+					name: "config",
+					config() {
+						return {
+							ssr: {
+								// externalize cjs or native deps
+								external: [
+									"@node-rs/argon2",
+									"better-sqlite3",
+								],
+							},
+						};
+					},
+				},
+			],
+		}),
+	],
+	optimizeDeps: {
+		// silence esbuild deps optimization warning
+		exclude: ["@node-rs/argon2"],
+	},
```

Comment:
This is to silence client pre-bundling warning:

```
✘ [ERROR] Could not resolve "@node-rs/argon2-wasm32-wasi"

    node_modules/.pnpm/@node-rs+argon2@1.8.3/node_modules/@node-rs/argon2/browser.js:1:14:
      1 │ export * from '@node-rs/argon2-wasm32-wasi'
        ╵               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

  You can mark the path "@node-rs/argon2-wasm32-wasi" as external to exclude it from the bundle,
  which will remove this error and leave the unresolved path in the bundle.
```

Reply:
If we find an external heuristics, then we can apply the same for client config to avoid this warning.